riposte-core/src/test/java/com/nike/riposte/server/componenttest/VerifyDecoderFailedResultIsHandledTest.java

```diff
@@ -0,0 +1,203 @@
+import com.nike.riposte.server.http.ProxyRouterEndpoint;
+import com.nike.riposte.server.http.RequestInfo;
+import com.nike.riposte.server.http.ResponseInfo;
+import com.nike.riposte.server.http.StandardEndpoint;
+import com.nike.riposte.server.testutils.ComponentTestUtils;
+import com.nike.riposte.util.Matcher;
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.http.DefaultFullHttpRequest;
+import io.netty.handler.codec.http.HttpHeaders;
+import io.netty.handler.codec.http.HttpMethod;
+import io.netty.handler.codec.http.HttpRequest;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.netty.handler.codec.http.HttpVersion;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+
+import static com.nike.riposte.server.testutils.ComponentTestUtils.executeRequest;
+import static com.nike.riposte.util.AsyncNettyHelper.supplierWithTracingAndMdc;
+import static java.lang.Thread.sleep;
+import static java.util.Collections.singleton;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class VerifyDecoderFailedResultIsHandledTest {
+
+    private static Server proxyServer;
+    private static ServerConfig proxyServerConfig;
+    private static Server downstreamServer;
+    private static ServerConfig downstreamServerConfig;
+    private static final int incompleteCallTimeoutMillis = 5000;
+
+    @BeforeClass
+    public static void setUpClass() throws Exception {
+        downstreamServerConfig = new DownstreamServerTestConfig();
+        downstreamServer = new Server(downstreamServerConfig);
+        downstreamServer.startup();
+
+        proxyServerConfig = new ProxyTestingTestConfig();
+        proxyServer = new Server(proxyServerConfig);
+        proxyServer.startup();
+    }
+
+    @AfterClass
+    public static void tearDown() throws Exception {
+        proxyServer.shutdown();
+        downstreamServer.shutdown();
+    }
+
+    @Test
+    public void proxy_endpoints_should_handle_decode_exception() throws Exception {
+        // given
+        int payloadSize = 10000;
+        ByteBuf payload = ComponentTestUtils.createByteBufPayload(payloadSize);
+
+        HttpRequest request = new DefaultFullHttpRequest(
+                HttpVersion.HTTP_1_1, HttpMethod.POST, RouterEndpoint.MATCHING_PATH, payload
+        );
+
+        //leave off content-length and transfer-encoding to trigger DecoderFailedResult
+        request.headers().set(HttpHeaders.Names.HOST, "localhost");
+
+        // when
+        Pair<Integer, String> serverResponse = executeRequest(request, proxyServerConfig.endpointsPort(), incompleteCallTimeoutMillis);
+
+        // then
+        assertThat(serverResponse.getLeft()).isEqualTo(HttpResponseStatus.BAD_REQUEST.code());
```

Comment:
This assertion is a little thin for verifying that the server threw the correct exception. Can you do something like [this](https://github.com/Nike-Inc/riposte/blob/f057c68f01fa67c143441de7d6fe4369c9d527b8/riposte-core/src/test/java/com/nike/riposte/server/componenttest/VerifyMiscellaneousFunctionalityComponentTest.java#L65) instead, including error metadata verification?